feat(minifier): compress `if (foo, !!bar)` to `if (foo, bar)`

sapphi-red opened the pull request:
`if (foo, !!bar)` can be compressed to `if (foo, bar)` because the last value is used as a boolean.

Boshen commented:
### Merge activity

* **Apr 1, 9:06 AM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

sapphi-red committed:
feat(minifier): compress `if (foo, !!bar)` to `if (foo, bar)` (#10164)

`if (foo, !!bar)` can be compressed to `if (foo, bar)` because the last value is used as a boolean.
